elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -45,24 +45,38 @@ export interface P2PSharedState {
 	// tslint:disable-next-line: no-mixed-interface
 	readonly [key: string]: unknown;
 }
-
+// Disable readonly properties as its going to change
+// tslint:disable:readonly-keyword
 export interface P2PInternalState {
-	readonly dateAdded?: Date;
-	readonly peerKind?: PeerKind;
-	readonly isBanned?: boolean;
-	readonly productivity?: number;
-	readonly reputation?: number;
-	readonly connectionKind?: ConnectionKind;
-	readonly advertiseAddress: boolean;
-}
-
+	dateAdded?: Date;
+	peerKind: PeerKind;
+	productivity: {
+		requestCounter: number;
+		responseCounter: number;
+		responseRate: number;
+		lastResponded: number;
+	};
+	reputation: number;
+	netgroup: number;
+	latency: number;
+	connectTime: number;
+	messageCounter: Map<string, number>;
+	messageRates: Map<string, number>;
+	rpcCounter: Map<string, number>;
+	rpcRates: Map<string, number>;
+	wsMessageCount: number;
+	wsMessageRate: number;
+	connectionKind: ConnectionKind;
+	advertiseAddress: boolean;
+}
+// tslint:enable:readonly-keyword
 export interface P2PPeerInfo {
 	// String to uniquely identify each peer
 	readonly peerId: string;
 	readonly ipAddress: string;
 	readonly wsPort: number;
 	readonly sharedState?: P2PSharedState;
-	readonly internalState?: P2PInternalState;
+	readonly internalState?: Partial<P2PInternalState>;
```

Comment:
In this way, we will assign a lot of values and empty maps to a `peerInfo` which we may not connect for a long time for example in `newPeers` list.

Reply:
We just need `ConnectedPeerInfo` within the Base peer class to impose mandatory `internalState` value, that we can't impose at the beginning. Because whenever we receive any new peer, we sanitize and take its `sharedState` values but we cannot check-in util functions whether it is present in peerBook or not